server/service/src/settings_service.rs

```diff
@@ -99,6 +99,11 @@ pub trait SettingsServiceTrait: Sync + Send {
             .map_err(|err| UpdateSettingsError::RepositoryError(err.to_inner_error()))?;
         Ok(result)
     }
+
+    fn disable_sync(&self, ctx: &ServiceContext) -> Result<(), RepositoryError> {
```

Comment:
for the other key value pairs I always kept the setters and getters close together, so maybe move the the setter here as well?

Just noticed that all the `KeyValueType::SettingsSync*` values are for populating the `SyncSettings` struct but I don't think you want to but the disabled flag into the SyncSettings, right? maybe rename `KeyValueType::SettingsSyncIsDisabled` to `KeyValueType::SyncDisabled`?

(I called it KeyValueType::SettingsSync* instead of KeyValueType::SyncSettings* to have all the settings key value pairs start with Settings*... not strong about it)

Reply:
Hmm, I kind of want all sync settings to be in SyncSettings, I think they can all be grouped by `SyncSettings` (maybe sync site id is an exemption, but sync disabled should be changeable by user via UI).

> I called it KeyValueType::SettingsSync* instead of KeyValueType::SyncSettings* to have all the settings key value pairs start with Settings*... not strong about it)

I like `SyncSettings` to be honest. It's very similar to discussions we had about partitioning by functional area vs characteristic. (i.e. having all component for front end in components folder vs having a functional area like home page which has sub directory for components for home), also similar to how we name datetime types in db, they don't start with datetime_ (characteristic) but with their functional part (created_)

I'll make an issue before merging, but what I think should happen:

* We use SyncSettings from database for sync (don't pass them from server startup), I think V5 will remove sync_site_id from requests, so we wouldn't need to grab it from db at this point.
* If env vars exist on startup for sync settings, but settings already exist in db, panic! while informing user that site is already initialisng.

Reply:
sounds good